src/main/java/bio/terra/service/policy/PolicyService.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PolicyService {
+  private static final Logger logger = LoggerFactory.getLogger(PolicyService.class);
+  private final PolicyServiceConfiguration policyServiceConfiguration;
+
+  @Autowired
+  public PolicyService(PolicyServiceConfiguration policyServiceConfiguration) {
+    this.policyServiceConfiguration = policyServiceConfiguration;
+    logger.info("TPS base path: '{}'", policyServiceConfiguration.getBasePath());
+  }
+
+  // -- Policy Attribute Object Interface --
+  public void createSnapshotPao(UUID snapshotId, @Nullable TpsPolicyInputs policyInputs) {
+    createPao(snapshotId, TpsObjectType.SNAPSHOT, policyInputs);
+  }
+
+  public void createPao(
+      UUID resourceId, TpsObjectType resourceType, @Nullable TpsPolicyInputs policyInputs) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsPolicyInputs inputs = (policyInputs == null) ? new TpsPolicyInputs() : policyInputs;
+
+    TpsApi tpsApi = policyApi();
+    try {
+      tpsApi.createPao(
+          new TpsPaoCreateRequest()
+              .objectId(resourceId)
+              .component(TpsComponent.TDR)
+              .objectType(resourceType)
+              .attributes(inputs));
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  public void deletePao(UUID resourceId) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsApi tpsApi = policyApi();
+    try {
+      try {
+        tpsApi.deletePao(resourceId);
+      } catch (ApiException e) {
+        throw convertApiException(e);
+      }
+    } catch (PolicyServiceNotFoundException e) {
+      // Not found should not cause a failure on delete.
+    }
+  }
+
+  private ApiClient getApiClient(String accessToken) {
+    ApiClient client = new ApiClient();
+    client.setAccessToken(accessToken);
+    return client;
+  }
+
+  private ApiClient getUnAuthApiClient() {
+    ApiClient client = new ApiClient();
+    client.setBasePath(policyServiceConfiguration.getBasePath());
+    return client;
+  }
+
+  private TpsApi policyApi() {
+    try {
+      return new TpsApi(
+          getApiClient(policyServiceConfiguration.getAccessToken())
+              .setBasePath(policyServiceConfiguration.getBasePath()));
+    } catch (IOException e) {
+      throw new PolicyServiceAuthorizationException(
+          "Error reading or parsing credentials file", e.getCause());
+    }
+  }
+
+  @VisibleForTesting
+  public RuntimeException convertApiException(ApiException ex) {
+    if (ex.getCode() == HttpStatus.UNAUTHORIZED.value()) {
+      return new PolicyServiceAuthorizationException(
+          "Not authorized to access Terra Policy Service", ex.getCause());
+    } else if (ex.getCode() == HttpStatus.NOT_FOUND.value()) {
+      return new PolicyServiceNotFoundException("Policy service returns not found exception", ex);
```

Comment:
Double-checking, do these exception wrappers communicate what we want to communicate to users in the event of an exception?

Reply:
Do you think something like `PolicyAccessObjectNotFoundException` is clearer? I borrowed this from the workspace manager and initially thought this meant the policy service itself could not be found (then I realized it's just your typical not found exception coming from the policy service).

Reply:
Ohhh sorry, I should have been more clear. I meant that as a comment on the exception messages!